src/qibolab/instruments/qm/controller.py

```diff
@@ -179,7 +201,12 @@ def connect(self):
         """Connect to the Quantum Machines manager."""
         host, port = self.address.split(":")
         octave = declare_octaves(self.octaves, host, self.calibration_path)
-        self.manager = QuantumMachinesManager(host=host, port=int(port), octave=octave)
+        credentials = None
+        if self.cloud:
+            credentials = create_credentials()
```

Comment:
Thanks, that's interesting, I don't know much about it either. Here it was just moved here from another existing file that I dropped (simulator.py). Initially it was also just copied from a tutorial. It is only needed when using a cloud simulator provided by QM, not for the real instrument. Btw this simulator is not working for a while now, so we could even drop, I just left it in case it comes back in the future (because it is useful for some quick schedule tests).

Reply:
I don't mind too much about keeping it, for the time being. It's already there and quite a small overhead.

It was more of a curiosity.
(I was also interested in whether we could hijack some data and use the simulator in the CI, without having access to the instrument. But if the simulator is not working anyhow is useless)